### PR TITLE
JBPM-9489 PAM Authorisation Issue - REST Endpoints accessible to all …

### PR DESCRIPTION
**Pull request by bxf12315**
…logged in users - even those without the correct roles

**Thank you for submitting this pull request**

**JIRA**: _(please edit the JIRA link if it exists)_ 

[JBPM-9489](https://issues.redhat.com/browse/JBPM-9489)

**referenced Pull Requests**: _(please edit the URLs of referenced pullrequests if they exist)_

* paste the link(s) from GitHub here
* link 2
* link 3 etc.

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
Jenkins run fdb

**Comment by barboras7** [comment]
jenkins run fdb

**Comment by barboras7** [comment]
Hi @bxf12315,
I tested the business central and the issue seems to be fixed. However, I have a question about your fix. Why did you choose the roles that you chose (admin, rest-all, user)? I am especially concerned about the user role.

**Comment by bxf12315** [comment]
> Hi @bxf12315,
> I tested the business central and the issue seems to be fixed. However, I have a question about your fix. Why did you choose the roles that you chose (admin, rest-all, user)? I am especially concerned about the user role.

the user role is unnecessary, I will remove it.

**Comment by bxf12315** [comment]
jenkins run fdb

**Review by barboras7 — APPROVED** [review]

File: business-central-parent/business-central-webapp/src/main/webapp/WEB-INF/web.xml [hunk 1]
```diff
@@ -433,6 +433,21 @@ See http://www.w3.org/TR/SVG/intro.html#MIMEType. -->
...
+      <web-resource-name>rest controller</web-resource-name>
+      <url-pattern>/rest/controller/*</url-pattern>
+      <http-method>GET</http-method>
+      <http-method>PUT</http-method>
+      <http-method>POST</http-method>
+      <http-method>DELETE</http-method>
+    </web-resource-collection>
+    <auth-constraint>
+      <role-name>admin</role-name>
+      <role-name>rest-all</role-name>
```
Comment (caponetto) [inline business-central-parent/business-central-webapp/src/main/webapp/WEB-INF/web.xml:447]: Just to double-check: we don't need the `kie-server` role here too, right?

**Review by caponetto — COMMENTED** [review]
Hi @bxf12315, thanks for the PR!
I've left a comment, please check if it makes sense.

File: business-central-parent/business-central-webapp/src/main/webapp/WEB-INF/web.xml (hunk 1 above)
Comment (bxf12315) [reply business-central-parent/business-central-webapp/src/main/webapp/WEB-INF/web.xml:447]: where is kie-server role? It just BC's role in here.

Comment (caponetto) [reply business-central-parent/business-central-webapp/src/main/webapp/WEB-INF/web.xml:447]: Yes, this role is not here. This is exactly what I want to double-check.

**Review by caponetto — APPROVED** [review]
Since `kie-server` role is not necessary, it looks good to me.

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by pefernan** [comment]
jenkins retest this please

**Comment by bxf12315** [comment]
Jenkins retest this

**Comment by pefernan** [comment]
Jenkins run fdb

**Comment by pefernan** [comment]
Jenkins run fdb

**Pull request merged**